CMakeLists.txt

```diff
@@ -55,10 +55,10 @@ add_custom_target(tests)
 enable_testing()
 
 # PEGTL
-set(PEGTL_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/pegtl)
-set(PEGTL_INCLUDE_DIR ${PEGTL_ROOT}/include)
+set(TAO-PEGTL_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/pegtl)
+set(PEGTL_INCLUDE_DIR ${TAO-PEGTL_ROOT}/include)
 set(PEGTL_BUILD_TESTS OFF CACHE BOOL "")
-add_subdirectory(${PEGTL_ROOT} EXCLUDE_FROM_ALL)
+add_subdirectory(${TAO-PEGTL_ROOT} EXCLUDE_FROM_ALL)
```

Comment:
I don't understand this change. Could you elaborate?

Reply:
Hm weird... I already answered but it some where lost :-( ... ok in short again:

It is possible to import the third party library as `target` via `pkg_check_modules()` with the `IMPORTED_TARGET`.

The big advantage about the imported target is to use the transitive usage requirements from the target. This means, you should only add the third party library as link library, for example the botan libary:
```
target_link_libraries(foo ...
   ...
   PkgConfig::botan-2
   ...
 )
```
You didn't need to add the include path via `target_include_directories()` or the compile definitions via `target_compile_definitions()`. The include path and compile definitons are set via the imported target `PkgConfig::botan-2` on `target_link_libraries()`. As additional, botan has dependencies to BZip2 and ZLib and these libraries will be added also to the linker at link time.

See also in the cmake documentation [imported targets](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets) and [transitive usage requirements](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#transitive-usage-requirements)
Or I also recommended to show the talks from C++Now 2017: Daniel Pfeifer “Effective CMake" and CppCon 2017: Mathieu Ropert “Using Modern CMake Patterns to Enforce a Good Modular Design”. lambdafu also mentioned this here #83